addons/github/models.py

```diff
@@ -197,22 +197,19 @@ def to_json(self, user):
             valid_credentials = False
             owner = self.user_settings.owner
             connection = GitHubClient(external_account=self.external_account)
-            # TODO: Fetch repo list client-side
-            # Since /user/repos excludes organization repos to which the
-            # current user has push access, we have to make extra requests to
-            # find them
+
             valid_credentials = True
             try:
-                repos = itertools.chain.from_iterable((connection.repos(), connection.my_org_repos()))
-                repo_names = [
+                repos = itertools.chain.from_iterable((connection.repos(), connection.user_team_repos()))
+                repo_names = {
                     '{0} / {1}'.format(repo.owner.login, repo.name)
                     for repo in repos
-                ]
+                }
             except GitHubError:
                 repo_names = []
                 valid_credentials = False
             if owner == user:
-                ret.update({'repo_names': repo_names})
+                ret.update({'repo_names': list(repo_names)})
```

Comment:
This `set`/`list` conversion effectively dedupes it, but scrambles the ordering. You can use `sorted` to fix that.